uswds/uswds
# USWDS - Checkbox: Create high contrast styles for indeterminate checkboxes

# Summary

Enhances @lpsinger’s indeterminate checkbox styles with high contrast styles.

> [!IMPORTANT]
This PR ************************only affects styles.************************ There is currently no functionality implementation with indeterminate checkboxes. Clicking the checkbox **will** toggle the checkbox but **will** ******not****** change the icon.
> 

## Breaking change

This is not a breaking change.

## Related issue

Closes #5645

## Related pull requests

_Release note will be added for feature branch_

#5549
#5650 

## Preview link

[Standard checkbox story →](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/cm-checkbox-indeterminate-high-contrast/?path=/story/components-form-inputs-checkbox--default)
[Indeterminate checkboxes story →](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/cm-checkbox-indeterminate-high-contrast/?path=/story/components-form-inputs-checkbox--indeterminate)
[Indeterminate checkbox tiles story →](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/cm-checkbox-indeterminate-high-contrast/?path=/story/components-form-inputs-checkbox--indeterminate-tile)

## Problem statement

#5549 introduces indeterminate checkbox styles. When viewed in high contrast modes, the icon is not always visible and checkboxes can still be visibly toggled.

https://github.com/uswds/uswds/assets/25211650/8f00c15b-3596-48b4-a73a-a9d5d9597af1

## Solution

Use the `SelectedItem` high contrast token to maintain a visible indeterminate icon. This change also visually enhances checked checkbox styles as well.

Using `SelectedItem` will allow USWDS checkbox styles to reflect high contrast color customization set by the user.

### Develop:

Dark

![image](https://github.com/uswds/uswds/assets/25211650/cca2ff49-3500-4cf0-8d79-12f680440da1)

Light

![image](https://github.com/uswds/uswds/assets/25211650/b2db9580-1843-4a34-ac7f-e1dad8c111f1)


### using `SelectedItem`:

Dark

![image](https://github.com/uswds/uswds/assets/25211650/94c7fbd2-7997-46ca-ae88-0d696eff3708)


Light
![image](https://github.com/uswds/uswds/assets/25211650/2e149a11-f534-4f6f-94d7-e16f019f6d30)

### Radio buttons

I opted to not change the radio selected color due to its low color contrast in light high contrast themes

Develop:

![image](https://github.com/uswds/uswds/assets/25211650/d102a5fd-a46c-4981-ad75-a9d524083610)

`SelectedItem` :

![image](https://github.com/uswds/uswds/assets/25211650/e5d031fb-3aa8-44f6-8424-00e33552edae)

_Contrast Ratio: 1.3:1_

## Testing and review

1. Visit the indeterminate checkbox test story.
2. Activate high contrast mode.
3. Confirm check and indeterminate icons are visible in light and dark modes
4. Click the indeterminate checkbox.
5. Confirm indeterminate icon does not disappear.
6. Consider next steps for functionality implementations.

<details><summary><strong>Activating high contrast mode on chrome</strong></summary>
<p>

1. Open devTools
2. Open the menu using the three dot icon
3. Highlight `More tools` and select `Rendering`
4. Scroll down to find “Emulate CSS media feature `forced-colors` and activate.
5. You can also change light and dark themes using the “Emulate CSS media feature prefers-color-scheme” 

</p>
</details> 

### Testing checklist

- [ ]  Check is visible in light and dark high contrast themes
- [ ]  Indeterminate icon is visible in light and dark themes
